test/common/buffer/watermark_buffer_test.cc

```diff
@@ -431,22 +431,32 @@ TEST_F(WatermarkBufferTest, OverflowWatermarkDisabledOnVeryHighValue) {
 
   // Make sure the overflow threshold will be above std::numeric_limits<uint32_t>::max()
   Runtime::LoaderSingleton::getExisting()->mergeValues({{"envoy.buffer.overflow_multiplier", "3"}});
-  buffer1.setWatermarks((std::numeric_limits<uint32_t>::max() / 3) + 1);
+  const uint32_t high_watermark_threshold = (std::numeric_limits<uint32_t>::max() / 3) + 1;
+  buffer1.setWatermarks(high_watermark_threshold);
 
   // Add many segments instead of full uint32_t::max to get around std::bad_alloc exception
   const uint32_t segment_denominator = 128;
   const uint32_t big_segment_len = std::numeric_limits<uint32_t>::max() / segment_denominator + 1;
   const std::string big_segment_str = std::string(big_segment_len, 'a');
+  // Fill the buffer with more than uint32_t::max bytes
+  EXPECT_GT(static_cast<uint64_t>(segment_denominator) * big_segment_len,
+            std::numeric_limits<uint32_t>::max());
```

Comment:
In the other PR I hinted that this could be made more efficient by replacing the add with reserve(big_segment_len) followed by commit with the iovecs returned by reserve.  You'ld still get a buffer with the right number of slices in it, but you won't be copying any data into the actual slices.

Reply:
Thanks, it really improved the test duration!